### PR TITLE
accept any kind of collection for clump.collect and clump.flatten

### PR DESCRIPTION
**Pull request by fwbrasil**
@williamboxhall

### CODE REVIEW

File: src/main/scala/io/getclump/Clump.scala [hunk 1]
```diff
@@ -31,8 +31,9 @@ sealed trait Clump[+T] {
   def apply(): Future[T] = get.map(_.get)
 
   def getOrElse[B >: T](default: => B): Future[B] = get.map(_.getOrElse(default))
-
-  def list[B](implicit ev: T <:< List[B]): Future[List[B]] = get.map(_.toList.flatten)
+  
+  def flatten[B >: T](implicit cbf: CanBuildFrom[Nothing, Nothing, B]): Future[B] =
```
Comment (fwbrasil) [inline src/main/scala/io/getclump/Clump.scala]: @williamboxhall The old name `list` doesn't make much sense as the method accepts other collections now, but I din't like `flatten` as well. Do you have suggestions?

File: src/main/scala/io/getclump/Clump.scala [hunk 2]
```diff
@@ -143,13 +145,16 @@ private[getclump] class ClumpJoin[A, B](a: Clump[A], b: Clump[B]) extends Clump[
       }
 }
 
-private[getclump] class ClumpCollect[T](list: List[Clump[T]]) extends Clump[List[T]] {
-  val upstream = list
-  val downstream = Future.value(List())
+private[getclump] class ClumpCollect[T, C[_] <: Iterable[_]](clumps: C[Clump[T]])(implicit cbf: CanBuildFrom[C[Clump[T]], T, C[T]]) extends Clump[C[T]] {
+  val upstream = 
+    clumps.toList.asInstanceOf[List[Clump[T]]]
```
Comment (fwbrasil) [inline src/main/scala/io/getclump/Clump.scala]: I couldn't avoid this cast, if you see `scala.concurrent.Future.sequece`, it also uses a cast.

File: src/test/scala/io/getclump/ClumpApiSpec.scala [hunk 3]
```diff
@@ -201,9 +211,17 @@ class ClumpApiSpec extends Spec {
...
+      "list" in {
+        Await.result(Clump.value(List(1, 2)).list) ==== List(1, 2)
+      }
+      "set" in {
+        Await.result(Clump.value(Set(1, 2)).list) ==== Set(1, 2)
+      }
+      "seq" in {
+        Await.result(Clump.value(Seq(1, 2)).list) ==== Seq(1, 2)
+      }
+      // Clump.value(1).flatten //doesn't compile
```
Comment (williamboxhall) [inline src/test/scala/io/getclump/ClumpApiSpec.scala]: comment should also be "list"

**Comment by williamboxhall** [comment]
:+1: but I also feel traverse could take an collection type too

**Comment by fwbrasil** [comment]
I couldn't get `traverse` working with `CanBuildFrom` easily, lets use a separate pull request to change it.

**Pull request merged**